### PR TITLE
Aggregate Pattern initial implementation

### PR DESCRIPTION
**Pull request by rhkp**
Hi Mriganka,
Please review and accept the Aggregate Pattern implementation PR.

Thanks,
kp

### CODE REVIEW

**Comment by mpaulgreen** [comment]
@rhkp The only information that is required is router configuration for three routers and firewall rules to implement. It needs to be here - 
https://github.com/RHEcosystemAppEng/amq-cob/tree/master/manualconfig/routertopology/aggregator. Also, installation Instruction of Interconnect router needs to be here 
https://github.com/RHEcosystemAppEng/amq-cob/tree/master/manualconfig/routertopology.

All other information like vagrant file, logging into vms and other broker configuration can be removed. Once you have done a cleanup, I will take a look at the router Config in detail.

**Comment by rhkp** [comment]
@mpaulgreen Committed with changes in the feedback. Please check and advise.

**Comment by mpaulgreen** [comment]
Installation readme - 

@rhkp - Some of the RHEL VMs you need to enable the following packages
sudo subscription-manager repos --enable=interconnect-2-for-rhel-8-x86_64-rpms
sudo subscription-manager repos --enable=amq-clients-2-for-rhel-8-x86_64-rpms

Installation of py client is not required in the installation step.

Configuration readme -
installation of nano and pre-requisite is not required.
installation of nano and ssh to consumer is not required.

@sgahlot, once this version is ready give it a test and once it is fixed, I can merge the PR

**Comment by rhkp** [comment]
@mpaulgreen & @sgahlot , Please check now and advise.

**Review by mpaulgreen — COMMENTED** [review]
@rhkp I think you missed to removed the pre-requisite in router configuration as mentioned in the last comment. Also the waypoint is not defined as yes for Producer and it is yes for consumer. .  I am reading the documentation here https://access.redhat.com/documentation/en-us/red_hat_amq/7.7/html-single/using_amq_interconnect/index#configuring-brokered-messaging-router-rhel. "A waypoint address identifies a queue on a broker to which you want to route messages." So, by not specifying a waypoint the configuration for consumer flow of messages needs to work. Can you please test the configuration with no waypoint in consumer?

**Comment by rhkp** [comment]
@mpaulgreen Sure. Removed the following prerequisites from Router Config
>>>
## Prerequisites
* 3 VMs each for Producer, Consumer and Aggregate Routers are available with RHEL8, AMQ Interconnect installed and RHEL8 subscriptions activated
* Two Brokers are configured in a single VM or two separate VMs as appropriate with their amqp ports opened from firewall atleast in a cluster mode.
>>>

Checking the removal of waypoint in consumer and test the same. Will share.

**Comment by sgahlot** [comment]
@rhkp do you want to change the ports back to `61616` and `61617` for broker-1 and broker-2 (in router config) as that's how we're running those (on a single VM - from either HA replication or HA shared store). Or, I can change it when I test this out

**Comment by rhkp** [comment]
@sgahlot Okay sure let me see if there are not much of conflict in my other configs I will try changing it.

**Comment by sgahlot** [comment]
@rhkp I think that's all I could remember (based on your demo yesterday). Rest I can see when I test your config out on my side.

**Comment by rhkp** [comment]
@mpaulgreen removal of Waypoint from Consumer does not work. Receiver does not receive messages. However once we add Waypoint again and restart router, messages are coming up.

**Comment by rhkp** [comment]
@sgahlot The Router configs do not use these ports so we should be good.

**Comment by mpaulgreen** [comment]
@rhkp thanks for testing. Please push last changes. @sgahlot once you have tested it after @rhkp has pushed the last changs. Let me know I can merge the PR.

**Comment by rhkp** [comment]
@mpaulgreen @sgahlot Done. Please check and advise.

**Comment by sgahlot** [comment]
@rhkp @mpaulgreen let me setup 4 VMs for this and then test it on my local box.

**Comment by mpaulgreen** [comment]
@sgahlot cant u use your setup that you are performing for fanout pattern?

**Comment by rhkp** [comment]
@sgahlot If it may make it easier, I can share a Vagrant file which can easily bring up VMs on local with the IPs, unless this is being tested in IBM Cloud.

**Comment by sgahlot** [comment]
I had only setup 2 VMs so far (to test out the router->broker connectivity), but it's easy to do 2 more. I'm going to use my own config (for brokers and machines) but will use the config for the router from this PR

**Comment by rhkp** [comment]
Sure, please use whatever makes it easier for you.

File: manualconfig/routertopology/aggregator/readme.md [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+autoLink {
+    address: SampleQueue
+    connection: broker-connector-2
+    direction: out
+}
+```
+
+* As the Aggregate is going to listen for incoming connections from producer and consumer, open the needed ports
+```shell
+sudo firewall-cmd --permanent --add-port={5001,5772,5773,5673}/tcp
```
Comment (sgahlot) [inline manualconfig/routertopology/aggregator/readme.md:112]: @rhkp what's the importance of port 5001? I can't see it anywhere in the Aggregator config.

**Comment by rhkp** [comment]
@sgahlot I guess it may have lingered over from my original config then. I will remove that one. It can confuse us later.

**Comment by rhkp** [comment]
Okay, removed it.

**Review by mpaulgreen — COMMENTED** [review]
Thanks @rhkp and @sgahlot. As we are picking up new skills on this initiative, I think I will couple of more PRs. FYI @ccustine

**Pull request merged**